impl/src/test/java/io/kroxylicious/testing/kafka/common/KafkaClusterFactoryTest.java

```diff
@@ -26,7 +25,12 @@ void shouldThrowInContainerModeWithControllerOnlyNodes() {
                 .build();
 
         // When
-        assertThrows(IllegalStateException.class, () -> KafkaClusterFactory.create(kafkaClusterConfig));
+        try (var kafkaCluster = KafkaClusterFactory.create(kafkaClusterConfig)) {
+            assertThat(kafkaCluster).isNotNull();
+        }
+        catch (Exception e) {
```

Comment:
the catch is redundant, tests that throw uncaught exceptions fail anyway.  This pattern just needlessly wraps the exception.